### PR TITLE
Feature/chat

### PR DESCRIPTION
**Pull request by KambojRajan**
Feature: Chat component v1

This PR introduces the first version of the Chat Component

Added Features:
Auto-scroll: The chat automatically scrolls to display new messages.
Deal Lock Messages: This feature allows locking of deal-related messages.
Real-time Updates: Implemented real-time updates using Pusher (which uses WebSockets under the hood).
Unread Messages Count: Integrated with Avien Provider and Redis to keep track of unread messages.

Pending Fixes:
Infinite Load Issue: There’s an existing issue with the initial message fetch from the pipeline in the infinite load feature. This needs to be addressed.
Chat Lock in DB: The chat lock functionality in the database is not working and needs some checks.

### CODE REVIEW

**Comment by aryansri-19** [comment]
image
a little alignment fix

image
and when the current user locks a deal, display 'deal pending' instead of yes and no

**Comment by KambojRajan** [comment]
if a user lock a deal then why to display deal pending?

**Comment by KambojRajan** [comment]
> image a little alignment fix
> 
> image and when the current user locks a deal, display 'deal pending' instead of yes and no

**Comment by aryansri-19** [comment]
> @aryansri-19 and what is "a little alignment fix" what alignment u want?

the two tabs and the cross margin

**Comment by KambojRajan** [comment]
> > @aryansri-19 and what is "a little alignment fix" what alignment u want?
> 
> the two tabs and the cross margin
do I have to take the tabs a little down

**Comment by KambojRajan** [comment]
doing in 5 min

**Comment by aryansri-19** [comment]
> > > @aryansri-19 and what is "a little alignment fix" what alignment u want?
> > 
> > 
> > the two tabs and the cross margin
> > do I have to take the tabs a little down

yup 

and can the circle be fully round, like when the number is not that big inside it
image

currently it takes oval shape

**Comment by KambojRajan** [comment]
@aryansri-19 janab aap zara iss pr ko review karenge, isme sab implement ho chuka hai zara sab check kar lijiye ga, 
like
invite, ui chat ui, working of everything and I hope abb koi bhi changes ka gunjaish nahi hoga

**Comment by aryansri-19** [comment]
> Member

abhi karta hun

**Review by aryansri-19 — APPROVED** [review]
huh?

**Review by geekysilento — APPROVED** [review]
image

**Comment by KambojRajan** [comment]
this is bad, selfish

File: package.json [hunk 1]
```diff
@@ -37,19 +38,30 @@
...
     "@tailwindcss/container-queries": "^0.1.1",
+    "@tanstack/react-query": "^5.24.1",
     "@uploadthing/react": "^6.2.0",
+    "@upstash/redis": "^1.29.0",
+    "axios": "^1.6.7",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.0",
     "cmdk": "^0.2.1",
     "date-fns": "^3.3.1",
+    "dotenv": "^16.4.5",
```
Comment (heydoyouknowme0) [inline package.json:53]: chahiye?

File: app/components/BottomNav/index.tsx [hunk 2]
```diff
@@ -1,26 +1,20 @@
...
 import ButtonContainer from "./ButtonContainer";
 import NavButton from "./NavButton";
 import PlusBottomContainer from "./PlusBottomContainer";
 
 import CategoriesDrawer from "./CategoriesDrawer";
+import { auth } from "@/auth";
 
-const BottomNav = () => {
+const BottomNav = async () => {
+  const userData = await auth();
```
Comment (heydoyouknowme0) [inline app/components/BottomNav/index.tsx:12]: use toh nai hora

File: app/components/BottomNav/index.tsx [hunk 3]
```diff
@@ -1,26 +1,20 @@
...
-              viewBox="0 0 16 16"
-            >
-              <path d="m 6.6312013,0.0081728 c -4.4189192,0 -4.6555293,0.00271 -4.7971137,0.033203 -0.87742244,0.1889619 -1.55358294,0.9103611 -1.76481294,1.8828126 -0.066535,0.3063121 -0.071724,0.5594235 -0.06562,3.2792968 0.00629,2.8023458 0.00225,2.6580757 0.1087899,3.0371098 0.242768,0.863674 0.8450863,1.480167 1.65947674,1.699218 0.1095015,0.02946 0.2742242,0.03201 2.3985568,0.03711 l 2.2811328,0.0059 1.2933902,1.460938 c 0.7113043,0.803367 1.3158583,1.473604 1.3451943,1.490234 0.02933,0.01663 0.08643,0.03627 0.126059,0.04297 0.146832,0.02487 0.281309,-0.03249 0.388534,-0.164063 0.117003,-0.143612 0.113972,-0.10123 0.113972,-1.546875 v -1.28125 l 0.8271466,-0.0078 c 0.717301,-0.0059 0.842586,-0.01025 0.944573,-0.03711 0.916413,-0.241363 1.55875,-0.993609 1.733729,-2.0312496 0.02642,-0.15668 0.02941,-0.346633 0.03453,-2.7050782 0.006,-2.7195568 8.84e-4,-2.973147 -0.06562,-3.2792968 C 12.981893,0.951721 12.30575,0.2303407 11.428314,0.0413759 11.286729,0.0108839 11.05012,0.0081728 6.6312013,0.0081728 Z" />
-            </svg>
-          </NavButton>
-        </UserChat>
+        <UserChat />
```
Comment (heydoyouknowme0) [inline app/components/BottomNav/index.tsx:17]: I think merge ke time revert ho gya hai, the icons in normal navbar and mobile nav is different issliye aise hia

File: app/components/HomePage/Carousel/index.tsx [hunk 4]
```diff
@@ -41,7 +41,7 @@ export const CarouselItems = async ({
   productPageCarousel?: boolean;
 }) => {
   const data = await fetchRecentProducts();
-  return data!.map((product) => (
+  return data?.map((product) => (
```
Comment (heydoyouknowme0) [inline app/components/HomePage/Carousel/index.tsx:44]: .

**Pull request merged**